src/pages/deposit.vue

```diff
@@ -277,8 +264,21 @@ export default Vue.extend({
         return false;
       }
     },
-    displayTokenUnlock(): boolean {
-      return this.chosenToken && !this.enoughAllowance;
+    maxAmount(): string {
+      return !this.chosenToken ? "0" : closestPackableTransactionAmount(this.chosenToken.rawBalance).toString();
+    },
+    singleColumnButtons(): boolean {
+      return (
+        (!this.inputtedAmount && this.chosenToken && this.chosenToken.symbol.length > 5) ||
+        (!!this.inputtedAmount && this.$options.filters!.formatToken(this.amountBigNumber, (this.chosenToken as ZkInBalance).symbol).length > 13)
+      );
+    },
+    zeroAllowance(): boolean | undefined {
```

Comment:
How can it return `undefined`?

If `allowance` is undefined, it will always return `false`. Otherwise, `allowance!.eq(BigNumber.from("0"))` would always return some boolean